### PR TITLE
Fixed partial variables extended bug and split PartialVarsEnvExtended test into two

### PR DESCRIPTION
**Pull request by jduraniglesias**
Changed the varsCopy variable to unconditionally copy e.variables. Split the partial variables extended test into TestPartialVarsEnv and TestPartialVarsExtendedEnv which tests, respectively, if internals are initialized and if sub environment is correctly copied. 

Fixes an edge case with env.PartialVars on an extended environment if the base was used for type checking already.

### CODE REVIEW

File: cel/env.go [hunk 1]
```diff
@@ -309,17 +309,14 @@ func (e *Env) Extend(opts ...EnvOption) (*Env, error) {
...
 		chkOptsCopy = append(chkOptsCopy, checker.ValidatedDeclarations(chk))
-	} else {
-		// If the type-checker has not been instantiated, ensure the unvalidated declarations are
-		// provided to the extended Env instance.
-		varsCopy = make([]*decls.VariableDecl, len(e.variables))
-		copy(varsCopy, e.variables)
 	}
+	//
+	varsCopy := make([]*decls.VariableDecl, len(e.variables))
+	copy(varsCopy, e.variables)
```
Comment (TristonianJones) [inline cel/env.go:318]: Hi Javi,

I think the issue here might be that the validated declarations are expected to contain the variable declarations already, but in a more optimized form. It's possible that this change won't cause any harm, but it's worth running some env benchmarks

File: repl/evaluator.go [hunk 2]
```diff
@@ -579,6 +585,17 @@ func (e *Evaluator) AddOption(opt Optioner) error {
 	return nil
 }
```
Comment (TristonianJones) [inline repl/evaluator.go:587]: nit: please add a doc comment to this method

File: test/bench/bench.go [hunk 3]
```diff
@@ -40,6 +40,20 @@ type Case struct {
 	Out ref.Val
 }
 
+type EvalCase struct {
```
Comment (TristonianJones) [inline test/bench/bench.go]: Perhaps call this an `DynamicEnvCase` since this is related to tests related to changes in environment? Also, please add a top-level comment.

File: test/bench/bench.go [hunk 4]
```diff
@@ -178,6 +192,53 @@ var (
 	}
 )
 
+var (
```
Comment (TristonianJones) [inline test/bench/bench.go]: nit: you can place the variable below within the same `var` block as above.

File: test/bench/bench.go [hunk 5]
```diff
@@ -178,6 +192,53 @@ var (
...
+	ReferenceCheckerCases = []*EvalCase{
```
Comment (TristonianJones) [inline test/bench/bench.go]: nit: Consider renaming this to ReferenceDynamicEnvCases and add a comment to the exposed variable?

File: test/bench/bench.go [hunk 6]
```diff
@@ -188,6 +249,13 @@ func RunReferenceCases(b *testing.B, env *cel.Env) {
 	}
 }
 
+func RunReferenceCheckerCases(b *testing.B) {
```
Comment (TristonianJones) [inline test/bench/bench.go]: Please add a top-level comment for the exposed method.

File: test/bench/bench.go [hunk 7]
```diff
@@ -241,3 +309,17 @@ func RunCase(b *testing.B, env *cel.Env, bc *Case) {
 		})
 	}
 }
+
+func RunEvalCase(b *testing.B, bc *EvalCase) {
```
Comment (TristonianJones) [inline test/bench/bench.go]: nit: Please rename to `RunDynamicEnvCase` and add a top-level comment.

File: test/bench/bench.go [hunk 8]
```diff
@@ -40,6 +40,21 @@ type Case struct {
 	Out ref.Val
 }
 
+// Case represents an expression compiled in a dynamic environment.
```
Comment (TristonianJones) [inline test/bench/bench.go]: nit: the format for Go comments should match the type or function name, `DynamicEnvCase represents ...`

File: test/bench/bench.go [hunk 9]
```diff
@@ -188,6 +249,14 @@ func RunReferenceCases(b *testing.B, env *cel.Env) {
 	}
 }
 
+// RunReferenceCheckerCases evaluates the set of ReferenceDynamicEnvCases.
```
Comment (TristonianJones) [inline test/bench/bench.go]: nit: Can we rename this to match the test cases that are being run? i.e. `RunReferenceDynamicEnvCases`

**Review by TristonianJones — APPROVED** [review]
